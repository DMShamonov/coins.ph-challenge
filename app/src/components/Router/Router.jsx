// @flow

import * as React from 'react';
import { Switch, Route } from 'react-router-dom';

import DynamicImport from 'components/DynamicImport';
import ROUTES from 'constants/routes';
import ProtectedRoute from 'components/ProtectedRoute';

const AsyncSignInPage = DynamicImport({
  loader: () => import('components/SignInPage' /* webpackChunkName: 'sign-in-page' */),
});

const AsyncLayout = DynamicImport({
  loader: () => import('components/Layout' /* webpackChunkName: 'layout' */),
});

export default function Router(): React.Node {
  return (
    <Switch>
      <Route exact path={ROUTES.SIGN_IN} component={AsyncSignInPage} />
      <ProtectedRoute path={ROUTES.ROOT} component={AsyncLayout} />
    </Switch>
  );
}
