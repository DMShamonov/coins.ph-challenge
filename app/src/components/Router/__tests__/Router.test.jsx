import React from 'react';
import { shallow } from 'enzyme';

import Router from '../Router';

describe('`Router` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Router />)).toMatchSnapshot();
  });
});
