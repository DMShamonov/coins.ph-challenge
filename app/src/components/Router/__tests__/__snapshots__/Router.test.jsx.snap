// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Router\` component should render correctly with default props 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/sign-in"
  />
  <Connect(verifySession)
    component={[Function]}
    path="/"
  />
</Switch>
`;
